Content.Server/VendingMachines/VendingMachineSystem.cs

```diff
@@ -153,6 +165,28 @@ public void InitializeFromPrototype(EntityUid uid, VendingMachineComponent? vend
             vendComponent.Inventory = inventory;
         }
 
+        /// <summary>
+        /// Add more entries for any reason AFTER initialization (emag, machine upgrades, etc)
+        /// </summary>
+        public void AddVendEntries(VendingMachineComponent component, string pack)
+        {
+            if (!_prototypeManager.TryIndex(pack, out VendingMachineInventoryPrototype? packPrototype))
+            {
+                return;
+            }
+
+            var inventory = component.Inventory;
+            foreach (var (id, amount) in packPrototype.StartingInventory)
+            {
+                if (!_prototypeManager.TryIndex(id, out EntityPrototype? prototype))
+                {
+                    continue;
+                }
+                inventory.Add(new VendingMachineInventoryEntry(id, amount));
+            }
+            component.Inventory = inventory;
+        }
```

Comment:
This should be genericised rather than copying the above method (including bugs).
- Need to at least log on pack failure if it remains as TryIndex
- component.Inventory = inventory unnecessary for reference type.